app/models/vendor.rb

```diff
@@ -73,4 +73,17 @@ def status
       product_counts.key?(product_state) ? product_counts[product_state].count : 0
     end
   end
+
+  def header_fields?
+    url? || address? || !points_of_contact.empty?
+  end
+
+  def product_arr
+    Product.where(vendor_id: id).by_updated_at.order_by(state: 'desc')
+  end
+
+  def favorite_products(current_user)
+    product_arr_sorted = product_arr.sort_by { |a| (a.favorite_user_ids.include? current_user.id) ? 0 : 1 }
+    product_arr_sorted.select { |p| p.favorite_user_ids.include? current_user.id }
```

Comment:
I am having trouble understanding the purpose of the previous 2 lines. Mongo supports directly querying a database for the contents of an array, so replacing both the above lines with `product_arr.where(favorite_user_ids: current_user.id)` should actually work fine. I tested this locally with 2 separate users favoriting the same product and it seems to work fine to me. As you can see below even though favorite_user_ids has 2 elements in it, it still returned the product with multiple users favoriting properly.

```
product_arr.where(favorite_user_ids: current_user.id).first
=> #<Product _id: 5a21b467c60ac152505cbd7e, created_at: 2017-12-01 19:58:31 UTC, updated_at: 2017-12-01 20:24:13 UTC, vendor_id: BSON::ObjectId('5a21b3cec60ac151e4eda020'), bundle_id: BSON::ObjectId('5a1c3eb1c60ac172eeccafc6'), name: "test", version: "", description: "", c1_test: true, c2_test: false, c3_test: false, c4_test: false, randomize_records: true, duplicate_records: true, shift_records: false, allow_duplicate_names: false, measure_selection: "custom", measure_ids: ["40280381-4BE2-53B3-014B-E6419B2C033E"], favorite_user_ids: [BSON::ObjectId('5a1c3e86c60ac170d8a0f5d0'), BSON::ObjectId('5a21b95ec60ac152505cbda1')]>
```